realtek: add support for ZyXEL GS1920-24HPv1

andyboeh opened the pull request:
The GS1920-24HPv1 is a switch with 24 copper ports and 4 combo SFP/copper ports and PoE on the first 24 ports.

Specifications:
---------------
  * SoC: Realtek RTL8292M
  * Flash: 16 MiB SPI flash
  * RAM: 128 MiB
  * Ethernet: 24x 10/100/1000 Mbps
  * Buttons: 1x "Reset" button
  * UART: 1x serial header, standard DCE pinout (Tx = 2, Rx = 3, Gnd = 5); 9600 baud, 8n1, +- 5.6V logic levels
  * SFP: 4 combo copper/SFP ports
  * PoE: 24x
  * Fans: ADT7468 fan controller

Works:
------
  - (24) RJ-45 ethernet ports
  - Switch functions
  - Buttons
  - LEDs (partial support, the wrong LEDs light up)
  - Manual fan control

Not yet enabled:
----------------
  - PoE (requires patches to realtek-poe to support i2c)
  - Combo ports (link is up, but no data is transferred)

Fans:
-----
After boot, the fans are running in full speed mode. You can interact with the fan controller at /sys/class/hwmon/

Installation:
-------------

This device uses ZyNOS instead of Linux, this makes installation a bit more cumbersome. Serial console is required!

1. Set the switch to boot from the first image. This step is crucial, it will fail to boot if this is not set properly.

2. Connect to the switch using serial and interrupt the boot process to enter debug/recovery mode.

3. Load the OpenWrt initramfs image via XMODEM. You need to obtain an unlock code, based on your MAC address, first. See the excellent write up at https://www.ixo.de/info/zyxel_uclinux/ for details. Replace unlock_code in the commands below by the code obtained. After running ATBA5, the terminal needs to be closed and re-opened with 115200 baud. This speeds up the file transfer significantly! The file length in bytes need to be given instead of file_length below. You also need an XMODEM upload utility like "lrzsz-sx -X" to transfer the file. Start the XMODEM upload after running the ATUPxxxx command:

     > ATEN1,unlock_code
     > ATBA5
     > ATUP80100000,file_length
     > ATGO80100000

4. Wait for OpenWrt to boot. Once this is done, transfer the loader binary and the sysupgrade image to "/tmp" using scp.

5. Install OpenWrt permanently by running the following two commands:

    > mtd write /tmp/loader.bin loader
    > mtd write /tmp/sysupgrade.bin firmware

6. Reboot the switch and enjoy OpenWrt.

NB: You do not need to touch the loader binary unless it's recommended.
    The loader is not part of a regular sysupgrade file and will be left
    untouched. The boot loader only checks if the loader is valid to be
    able to boot.

Recovery/ Return to stock:
--------------------------

Just spam the "u" key during (or "z" for 9600 baud) during memory testing to trigger a recovery XMODEM upload at 115200 baud. A standard OEM upgrade image works properly.

andyboeh committed:
realtek: Makefile: add rt-loader-standalone and zynsig recipes

This adds the build recipes for rt-loader-standalone and zynsig to the
Makefile. The recipes are required for the upcoming ZyXEL GS1920 series.

Signed-off-by: Andreas Böhler <dev@aboehler.at>
Link: https://github.com/openwrt/openwrt/pull/20439
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

jonasjelonek commented:
Ahh no I was wrong. I actually mixed that up with another Zyxel device here. I cannot do `$` during boot and looking at the partition dump, there is nothing that hints to U-Boot. So no, this is running the normal bootbase.

Sorry for that!

andyboeh commented:
To be on the safe side, please dump the beginning of the flash and run "strings" on it. On the XGM1915E there was also *nothing* pointing towards U-Boot until we analyzed the flash dump.

Back to the topic: Great that a lot already works on the v2 - we will probably need a separate build recipe and/or we will have to fix the zynsig tool.

jonasjelonek commented:
Thanks for that, then I completely missed these details at some point ...

I have a flash clamp and did a full dump before so I can restore in worst case. But I didn't break it completely because second partition is still booted so all fine.

Then I have to adjust this for my device.

EDIT: So this should have no effect on this PR, this should be added in another PR.

andyboeh commented:
> So this should have no effect on this PR, this should be added in another PR.

Sure. I will only prepare for v1/v2, once this is merged, we can bring up yours. The GS2210 series and other members of the GS1920 series might also follow, but I only have a GS1920-24HP.

Neustradamus commented:
@andyboeh: Nice PR, good job!

jonasjelonek reviewed (approved):
Thanks for your work, especially the Zyxel Bootloader stuff!

andyboeh commented:
> @andyboeh Could you provide a full OpenWrt bootlog of your v1? I'd like to compare this to what I have on my v2 to sort out some issues. Luckily, adjusting the addresses regarding the partitions is enough and zynsig is working fine too.

Here you go:
<details>
<summary>dmesg</summary>

```
root@OpenWrt:~# dmesg
[    0.000000] Linux version 6.12.51 (andy@x13y) (mips-openwrt-linux-musl-gcc (OpenWrt GCC 14.3.0 r30883-9c895bacfb) 14.3.0, GNU ld (GNU Binutils) 2.44) #0 SMP Fri Oct 17 18:46:22 2025
[    0.000000] SoC Type: Realtek RTL8392M rev B (6290)
[    0.000000] printk: legacy bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019555 (MIPS 34Kc)
[    0.000000] MIPS: machine is Zyxel GS1920-24HP
[    0.000000] earlycon: ns16550a0 at MMIO 0x18002000 (options '9600n8')
[    0.000000] printk: legacy bootconsole [ns16550a0] enabled
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Using appended Device Tree.
[    0.000000] OF: reserved mem: Reserved memory: No reserved-memory node in the DT
[    0.000000] Detected 1 available secondary CPU(s)
[    0.000000] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] percpu: Embedded 12 pages/cpu s18384 r8192 d22576 u49152
[    0.000000] pcpu-alloc: s18384 r8192 d22576 u49152 alloc=12*4096
[    0.000000] pcpu-alloc: [0] 0 [0] 1 
[    0.000000] Kernel command line: earlycon
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes, linear)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes, linear)
[    0.000000] Writing ErrCtl register=0001a005
[    0.000000] Readback ErrCtl register=0001a005
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 32768
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=2, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000] 	Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] RCU Tasks Trace: Setting shift to 1 and lim to 1 rcu_task_cb_adjust=1 rcu_task_cpu_ids=2.
[    0.000000] NR_IRQS: 256
[    0.000000] rcu: srcu_init: Setting srcu_struct sizes based on contention.
[    0.000000] rtl83xx-clk: initialized, CPU 650 MHz, MEM 300 MHz (16 Bit DDR3), LXB 200 MHz
[    0.000000] clocksource: realtek_otto_timer: mask: 0xfffffff max_cycles: 0xfffffff, max_idle_ns: 38225208801 ns
[    0.000002] sched_clock: 28 bits at 3125kHz, resolution 320ns, wraps every 42949672800ns
[    0.106950] Calibrating delay loop... 432.53 BogoMIPS (lpj=2162688)
[    0.239239] pid_max: default: 32768 minimum: 301
[    0.313927] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.410335] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.528967] rcu: Hierarchical SRCU implementation.
[    0.592162] rcu: 	Max phase no-delay instances is 1000.
[    0.663503] smp: Bringing up secondary CPUs ...
[    0.725363] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.725444] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.725562] CPU1 revision is: 00019555 (MIPS 34Kc)
[    0.786750] Counter synchronization [CPU#0 -> CPU#1]: passed
[    1.112895] smp: Brought up 1 node, 2 CPUs
[    1.168704] Memory: 117388K/131072K available (7528K kernel code, 644K rwdata, 1708K rodata, 1312K init, 219K bss, 12868K reserved, 0K cma-reserved)
[    1.353821] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    1.483600] futex hash table entries: 512 (order: 2, 16384 bytes, linear)
[    1.581364] pinctrl core: initialized pinctrl subsystem
[    1.654668] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    1.733666] thermal_sys: Registered thermal governor 'step_wise'
[    1.760797] rtl83xx-clk clock-controller: rate setting enabled, CPU 400-850 MHz, MEM 300-300 MHz, LXB 200-200 MHz, OVERCLOCK AT OWN RISK
[    2.006760] clocksource: Switched to clocksource realtek_otto_timer
[    2.103848] NET: Registered PF_INET protocol family
[    2.168578] IP idents hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    2.265166] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 4096 bytes, linear)
[    2.375455] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    2.477706] TCP established hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    2.578728] TCP bind hash table entries: 1024 (order: 2, 16384 bytes, linear)
[    2.673023] TCP: Hash tables configured (established 1024 bind 1024)
[    2.757858] MPTCP token hash table entries: 256 (order: 0, 4096 bytes, linear)
[    2.853770] UDP hash table entries: 256 (order: 1, 8192 bytes, linear)
[    2.939931] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes, linear)
[    3.033116] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    3.111857] workingset: timestamp_bits=14 max_order=15 bucket_order=1
[    3.203508] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    3.280476] jffs2: version 2.2 (NAND) (SUMMARY) (ZLIB) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    3.425307] pinctrl-single 1b000004.pinmux: 32 pins, size 4
[    3.499492] pinctrl-single 1b0000e4.pinmux: 32 pins, size 4
[    3.578086] Serial: 8250/16550 driver, 2 ports, IRQ sharing disabled
[    3.664168] printk: legacy console [ttyS0] disabled
[    3.729349] 18002000.uart: ttyS0 at MMIO 0x18002000 (irq = 31, base_baud = 12500000) is a 16550A
[    3.845427] printk: legacy console [ttyS0] enabled
[    3.971725] printk: legacy bootconsole [early0] disabled
[    4.112014] printk: legacy bootconsole [ns16550a0] disabled
[    4.304835] brd: module loaded
[    4.350912] 3 fixed-partitions partitions found on MTD device spi0.0
[    4.434897] OF: Bad cell count for /soc/spi@1200/flash@0/partitions
[    4.517733] OF: Bad cell count for /soc/spi@1200/flash@0/partitions
[    4.600839] Creating 3 MTD partitions on "spi0.0":
[    4.664048] 0x000000000000-0x000000020000 : "bootbase"
[    4.734271] OF: Bad cell count for /soc/spi@1200/flash@0/partitions
[    4.817658] 0x000000020000-0x0000000b0000 : "reserved"
[    4.887561] 0x0000000b0000-0x000001000000 : "factory"
[    4.957206] 2 fixed-partitions partitions found on MTD device factory
[    5.042259] OF: Bad cell count for /soc/spi@1200/flash@0/partitions
[    5.125423] Creating 2 MTD partitions on "factory":
[    5.189788] 0x000000000000-0x000000010000 : "loader"
[    5.257518] 0x000000010000-0x000000f50000 : "firmware"
[    5.328056] 2 uimage-fw partitions found on MTD device firmware
[    5.406301] Creating 2 MTD partitions on "firmware":
[    5.471762] 0x000000000000-0x0000002f0000 : "kernel"
[    5.539957] 0x0000002f0000-0x000000f40000 : "rootfs"
[    5.608774] mtd: setting mtd6 (rootfs) as root device
[    5.675611] 1 squashfs-split partitions found on MTD device rootfs
[    5.757267] 0x000000510000-0x000000f40000 : "rootfs_data"
[    5.856112] mdio-rtl-otto 1b000000.switchcore:mdio-controller: probing RTL8390 family mdio bus
[    6.219579] realtek-otto-serdes-mdio 1b000000.switchcore:mdio-serdes: Realtek SerDes mdio bus initialized, 14 SerDes, 12 pages, 32 registers
[    6.386987] realtek-otto-pcs 1b000000.switchcore:pcs: Realtek PCS driver initialized
[    6.490226] Probing RTL838X eth device pdev: 81958e00, dev: 81958e10
[    6.603332] Found SoC ID: 8392: RTL8392M, family 8390
[    6.670298] Using MAC 00004c9eff7983d4
[    6.722020] i2c_dev: i2c /dev entries driver
[    6.786721] NET: Registered PF_INET6 protocol family
[    6.857913] Segment Routing with IPv6
[    6.906427] In-situ OAM (IOAM) with IPv6
[    6.958429] NET: Registered PF_PACKET protocol family
[    7.025904] 8021q: 802.1Q VLAN Support v1.8
[    7.149811] i2c-gpio i2c-gpio-0: Slow GPIO pins might wreak havoc into I2C/SMBus bus timing
[    7.260681] i2c-gpio i2c-gpio-0: using lines 540 (SDA) and 539 (SCL)
[    7.345576] i2c-gpio i2c-gpio-1: Slow GPIO pins might wreak havoc into I2C/SMBus bus timing
[    7.456376] i2c-gpio i2c-gpio-1: using lines 546 (SDA) and 545 (SCL)
[    7.540872] i2c-gpio i2c-gpio-2: Slow GPIO pins might wreak havoc into I2C/SMBus bus timing
[    7.651594] i2c-gpio i2c-gpio-2: using lines 547 (SDA) and 551 (SCL)
[    7.736199] i2c-gpio i2c-gpio-3: Slow GPIO pins might wreak havoc into I2C/SMBus bus timing
[    7.847156] i2c-gpio i2c-gpio-3: using lines 558 (SDA) and 557 (SCL)
[    7.932153] i2c-gpio i2c-gpio-4: using lines 530 (SDA) and 528 (SCL)
[    8.020798] sfp sfp-p25: Host maximum power 1.0W
[    8.083966] sfp sfp-p25: No tx_disable pin: SFP modules will always be emitting.
[    8.182815] sfp sfp-p26: Host maximum power 1.0W
[    8.246354] sfp sfp-p26: No tx_disable pin: SFP modules will always be emitting.
[    8.345054] sfp sfp-p27: Host maximum power 1.0W
[    8.407432] sfp sfp-p27: No tx_disable pin: SFP modules will always be emitting.
[    8.506099] sfp sfp-p28: Host maximum power 1.0W
[    8.568424] sfp sfp-p28: No tx_disable pin: SFP modules will always be emitting.
[    8.673609] In rtl83xx_vlan_setup
[    8.717351] UNKNOWN_MC_PMASK: 001fffffffffffff
[    9.826638] rtl83xx_enable_phy_polling:    f000000ffffff
[   10.946993] rtl83xx-switch switch@1b000000: configuring for fixed/internal link mode
[   11.049995] rtl83xx-switch switch@1b000000: Link is Up - 1Gbps/Full - flow control off
[   11.074331] rtl83xx-switch switch@1b000000 lan1 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:00] driver [Realtek RTL8218B (external)] (irq=POLL)
[   11.349659] rtl83xx-switch switch@1b000000 lan2 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:01] driver [Realtek RTL8218B (external)] (irq=POLL)
[   11.544519] rtl83xx-switch switch@1b000000 lan3 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:02] driver [Realtek RTL8218B (external)] (irq=POLL)
[   11.739590] rtl83xx-switch switch@1b000000 lan4 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:03] driver [Realtek RTL8218B (external)] (irq=POLL)
[   11.934722] rtl83xx-switch switch@1b000000 lan5 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:04] driver [Realtek RTL8218B (external)] (irq=POLL)
[   12.129896] rtl83xx-switch switch@1b000000 lan6 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:05] driver [Realtek RTL8218B (external)] (irq=POLL)
[   12.325227] rtl83xx-switch switch@1b000000 lan7 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:06] driver [Realtek RTL8218B (external)] (irq=POLL)
[   12.520598] rtl83xx-switch switch@1b000000 lan8 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:07] driver [Realtek RTL8218B (external)] (irq=POLL)
[   12.732219] rtl83xx-switch switch@1b000000 lan9 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:08] driver [Realtek RTL8218B (external)] (irq=POLL)
[   12.928026] rtl83xx-switch switch@1b000000 lan10 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:09] driver [Realtek RTL8218B (external)] (irq=POLL)
[   13.125038] rtl83xx-switch switch@1b000000 lan11 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:0a] driver [Realtek RTL8218B (external)] (irq=POLL)
[   13.321655] rtl83xx-switch switch@1b000000 lan12 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:0b] driver [Realtek RTL8218B (external)] (irq=POLL)
[   13.518344] rtl83xx-switch switch@1b000000 lan13 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:0c] driver [Realtek RTL8218B (external)] (irq=POLL)
[   13.715009] rtl83xx-switch switch@1b000000 lan14 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:0d] driver [Realtek RTL8218B (external)] (irq=POLL)
[   13.911700] rtl83xx-switch switch@1b000000 lan15 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:0e] driver [Realtek RTL8218B (external)] (irq=POLL)
[   14.108385] rtl83xx-switch switch@1b000000 lan16 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:0f] driver [Realtek RTL8218B (external)] (irq=POLL)
[   14.321511] rtl83xx-switch switch@1b000000 lan17 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:10] driver [Realtek RTL8218B (external)] (irq=POLL)
[   14.518282] rtl83xx-switch switch@1b000000 lan18 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:11] driver [Realtek RTL8218B (external)] (irq=POLL)
[   14.714973] rtl83xx-switch switch@1b000000 lan19 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:12] driver [Realtek RTL8218B (external)] (irq=POLL)
[   14.911609] rtl83xx-switch switch@1b000000 lan20 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:13] driver [Realtek RTL8218B (external)] (irq=POLL)
[   15.108251] rtl83xx-switch switch@1b000000 lan21 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:14] driver [Realtek RTL8218B (external)] (irq=POLL)
[   15.304968] rtl83xx-switch switch@1b000000 lan22 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:15] driver [Realtek RTL8218B (external)] (irq=POLL)
[   15.501604] rtl83xx-switch switch@1b000000 lan23 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:16] driver [Realtek RTL8218B (external)] (irq=POLL)
[   15.698270] rtl83xx-switch switch@1b000000 lan24 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:17] driver [Realtek RTL8218B (external)] (irq=POLL)
[   15.894983] Realtek RTL8214FC 1b000000.switchcore:mdio-controller-mii:30: power off copper
[   16.005492] Realtek RTL8214FC 1b000000.switchcore:mdio-controller-mii:30: power on fibre
[   16.114370] rtl83xx-switch switch@1b000000 lan25 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:30] driver [Realtek RTL8214FC] (irq=POLL)
[   16.299282] Realtek RTL8214FC 1b000000.switchcore:mdio-controller-mii:31: power off copper
[   16.409793] Realtek RTL8214FC 1b000000.switchcore:mdio-controller-mii:31: power on fibre
[   16.518635] rtl83xx-switch switch@1b000000 lan26 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:31] driver [Realtek RTL8214FC] (irq=POLL)
[   16.704013] Realtek RTL8214FC 1b000000.switchcore:mdio-controller-mii:32: power off copper
[   16.814505] Realtek RTL8214FC 1b000000.switchcore:mdio-controller-mii:32: power on fibre
[   16.923774] rtl83xx-switch switch@1b000000 lan27 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:32] driver [Realtek RTL8214FC] (irq=POLL)
[   17.109104] Realtek RTL8214FC 1b000000.switchcore:mdio-controller-mii:33: power off copper
[   17.219616] Realtek RTL8214FC 1b000000.switchcore:mdio-controller-mii:33: power on fibre
[   17.328564] rtl83xx-switch switch@1b000000 lan28 (uninitialized): PHY [1b000000.switchcore:mdio-controller-mii:33] driver [Realtek RTL8214FC] (irq=POLL)
[   17.510944] rtl838x-eth 1b00a300.ethernet eth0: entered promiscuous mode
[   17.599606] DSA: tree 0 setup
[   17.638872] LINK state irq: 21
[   17.679170] In rtl83xx_setup_qos
[   17.721684] Setting up RTL839X QoS
[   17.766684] RTL839X_PRI_SEL_TBL_CTRL(i): 01112111
[   17.828776] Current Intprio2queue setting: 00000000
[   17.893098] QM_PKT2CPU_INTPRI_MAP: 00fac688
[   17.948454] rtl838x_dbgfs_init called
[   17.948540] rtl83xx_fib_event_work_do: FIB4 default rule failed
[   18.075296] rtl83xx_fib_event_work_do: FIB4 default rule failed
[   18.153861] clk: Disabling unused clocks
[   18.215987] VFS: Mounted root (squashfs filesystem) readonly on device 31:6.
[   18.317539] Freeing unused kernel image (initmem) memory: 1312K
[   18.395659] This architecture does not have kernel memory protection.
[   18.480614] Run /sbin/init as init process
[   18.534558]   with arguments:
[   18.573665]     /sbin/init
[   18.609312]   with environment:
[   18.650695]     HOME=/
[   18.681750]     TERM=linux
[   19.444937] init: Console is alive
[   19.490389] init: - watchdog -
[   19.971650] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[   20.120470] gpio_button_hotplug: loading out-of-tree module taints kernel.
[   20.215996] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[   20.315341] init: - preinit -
[   22.906732] random: crng init done
[   25.915347] RESETTING 8390, CPU_PORT 52
[   26.167100] rtl838x-eth 1b00a300.ethernet eth0: configuring for fixed/internal link mode
[   26.273801] In rtl838x_mac_config, mode 1
[   26.327712] rtl83xx-switch switch@1b000000 lan1: configuring for phy/qsgmii link mode
[   26.431142] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 0 and sds 0 not yet fully implemented
[   26.573666] rtl838x-eth 1b00a300.ethernet eth0: Link is Up - 1Gbps/Full - flow control off
[   26.577058] 8021q: adding VLAN 0 to HW filter on device lan1
[   26.758783] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   26.845101] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   26.937245] rtl83xx-switch switch@1b000000: add IPv4 route 192.168.1.1/32 (VLAN 0, MAC 4c:9e:ff:79:83:d4)
[   27.063591] rtl83xx-switch switch@1b000000: lower interface lan1 not found
[   27.154289] rtl83xx-switch switch@1b000000: fib_add() failed
[   27.229840] rtl83xx-switch switch@1b000000: add IPv4 route 192.168.1.255/32 (VLAN 0, MAC 4c:9e:ff:79:83:d4)
[   27.358432] rtl83xx-switch switch@1b000000: skip loopback/broadcast addresses and default routes
[   27.475123] rtl83xx-switch switch@1b000000: add IPv4 route 192.168.1.0/24 (VLAN 0, MAC 4c:9e:ff:79:83:d4)
[   27.601539] rtl83xx-switch switch@1b000000: lower interface lan1 not found
[   27.692408] rtl83xx-switch switch@1b000000: fib_add() failed
[   28.786827] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   32.083944] jffs2: notice: (996) jffs2_build_xattr_subsystem: complete building xattr subsystem, 7 of xdatum (0 unchecked, 1 orphan) and 8 of xref (1 dead, 0 orphan) found.
[   32.290438] mount_root: switching to jffs2 overlay
[   32.362445] overlayfs: upper fs does not support tmpfile.
[   32.448595] urandom-seed: Seeding with /etc/urandom.seed
[   32.638362] rtl83xx-switch switch@1b000000: delete IPv4 route 192.168.1.0/24 (VLAN 0, MAC 4c:9e:ff:79:83:d4)
[   32.768268] rtl83xx-switch switch@1b000000: no such gateway: 0.0.0.0
[   32.852168] rtl83xx-switch switch@1b000000: fib_del() failed
[   32.927265] rtl83xx-switch switch@1b000000: delete IPv4 route 192.168.1.255/32 (VLAN 0, MAC 4c:9e:ff:79:83:d4)
[   33.059394] rtl83xx-switch switch@1b000000: skip loopback/broadcast addresses and default routes
[   33.187408] rtl83xx-switch switch@1b000000: delete IPv4 route 192.168.1.1/32 (VLAN 0, MAC 4c:9e:ff:79:83:d4)
[   33.198826] procd: - early -
[   33.317354] rtl83xx-switch switch@1b000000: no such gateway: 0.0.0.0
[   33.355973] procd: - watchdog -
[   33.439219] rtl83xx-switch switch@1b000000: fib_del() failed
[   34.279018] procd: - watchdog -
[   34.321934] procd: - ubus -
[   34.561434] procd: - init -
[   35.742953] kmodloader: loading kernel modules from /etc/modules.d/*
[   35.925948] hwmon_vid: Unknown VRM version of your CPU
[   35.994223] lm85 4-002e: Starting monitoring
[   36.237436] kmodloader: done loading kernel modules from /etc/modules.d/*
[   37.607672] urngd: v1.0.2 started.
[   38.863278] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   46.504223] in rtl838x_eth_stop
[   46.546903] rtl838x-eth 1b00a300.ethernet eth0: Link is Down
[   47.258024] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   47.344186] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   47.430474] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   47.516711] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   47.614774] RESETTING 8390, CPU_PORT 52
[   47.866549] rtl838x-eth 1b00a300.ethernet eth0: configuring for fixed/internal link mode
[   47.973236] In rtl838x_mac_config, mode 1
[   48.026980] rtl838x-eth 1b00a300.ethernet eth0: Link is Up - 1Gbps/Full - flow control off
[   48.028291] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   48.222443] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   48.321017] rtl83xx-switch switch@1b000000 lan1: configuring for phy/qsgmii link mode
[   48.424550] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 0 and sds 0 not yet fully implemented
[   48.568479] 8021q: adding VLAN 0 to HW filter on device lan1
[   48.650270] switch: port 1(lan1) entered blocking state
[   48.719511] switch: port 1(lan1) entered disabled state
[   48.788768] rtl83xx-switch switch@1b000000 lan1: entered allmulticast mode
[   48.879608] rtl838x-eth 1b00a300.ethernet eth0: entered allmulticast mode
[   48.970272] rtl83xx-switch switch@1b000000 lan1: entered promiscuous mode
[   49.086792] rtl83xx-switch switch@1b000000: add IPv4 route 192.168.1.1/32 (VLAN 1, MAC 4c:9e:ff:79:83:d4)
[   49.213351] rtl83xx-switch switch@1b000000: route hashtable extended for gw 0.0.0.0
[   49.316188] rtl83xx-switch switch@1b000000: add IPv4 route 192.168.1.255/32 (VLAN 1, MAC 4c:9e:ff:79:83:d4)
[   49.445124] rtl83xx-switch switch@1b000000: skip loopback/broadcast addresses and default routes
[   49.567554] rtl83xx-switch switch@1b000000: add IPv4 route 192.168.1.0/24 (VLAN 1, MAC 4c:9e:ff:79:83:d4)
[   49.694319] rtl83xx-switch switch@1b000000: route hashtable extended for gw 0.0.0.0
[   49.796277] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   49.882816] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   49.999195] rtl83xx-switch switch@1b000000 lan2: configuring for phy/qsgmii link mode
[   50.102868] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 1 and sds 0 not yet fully implemented
[   50.247195] 8021q: adding VLAN 0 to HW filter on device lan2
[   50.349781] switch: port 2(lan2) entered blocking state
[   50.418943] switch: port 2(lan2) entered disabled state
[   50.488360] rtl83xx-switch switch@1b000000 lan2: entered allmulticast mode
[   50.580200] rtl83xx-switch switch@1b000000 lan2: entered promiscuous mode
[   50.712577] rtl83xx-switch switch@1b000000 lan3: configuring for phy/qsgmii link mode
[   50.816249] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 2 and sds 0 not yet fully implemented
[   50.959521] 8021q: adding VLAN 0 to HW filter on device lan3
[   51.041096] switch: port 3(lan3) entered blocking state
[   51.110290] switch: port 3(lan3) entered disabled state
[   51.179571] rtl83xx-switch switch@1b000000 lan3: entered allmulticast mode
[   51.271320] rtl83xx-switch switch@1b000000 lan3: entered promiscuous mode
[   51.386024] rtl83xx-switch switch@1b000000 lan4: configuring for phy/qsgmii link mode
[   51.489409] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 3 and sds 0 not yet fully implemented
[   51.632533] 8021q: adding VLAN 0 to HW filter on device lan4
[   51.715526] switch: port 4(lan4) entered blocking state
[   51.784722] switch: port 4(lan4) entered disabled state
[   51.853942] rtl83xx-switch switch@1b000000 lan4: entered allmulticast mode
[   51.946236] rtl83xx-switch switch@1b000000 lan4: entered promiscuous mode
[   52.060553] rtl83xx-switch switch@1b000000 lan5: configuring for phy/qsgmii link mode
[   52.163946] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 4 and sds 1 not yet fully implemented
[   52.307400] 8021q: adding VLAN 0 to HW filter on device lan5
[   52.388362] switch: port 5(lan5) entered blocking state
[   52.457360] switch: port 5(lan5) entered disabled state
[   52.526672] rtl83xx-switch switch@1b000000 lan5: entered allmulticast mode
[   52.618252] rtl83xx-switch switch@1b000000 lan5: entered promiscuous mode
[   52.730948] rtl83xx-switch switch@1b000000 lan6: configuring for phy/qsgmii link mode
[   52.834348] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 5 and sds 1 not yet fully implemented
[   52.977604] 8021q: adding VLAN 0 to HW filter on device lan6
[   53.057896] switch: port 6(lan6) entered blocking state
[   53.126864] switch: port 6(lan6) entered disabled state
[   53.195859] rtl83xx-switch switch@1b000000 lan6: entered allmulticast mode
[   53.287429] rtl83xx-switch switch@1b000000 lan6: entered promiscuous mode
[   53.401229] rtl83xx-switch switch@1b000000 lan7: configuring for phy/qsgmii link mode
[   53.504591] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 6 and sds 1 not yet fully implemented
[   53.647789] 8021q: adding VLAN 0 to HW filter on device lan7
[   53.728174] switch: port 7(lan7) entered blocking state
[   53.797129] switch: port 7(lan7) entered disabled state
[   53.866218] rtl83xx-switch switch@1b000000 lan7: entered allmulticast mode
[   53.957756] rtl83xx-switch switch@1b000000 lan7: entered promiscuous mode
[   54.052567] rtl83xx-switch switch@1b000000 lan2: Link is Up - 1Gbps/Full - flow control rx/tx
[   54.071492] switch: port 2(lan2) entered blocking state
[   54.234096] switch: port 2(lan2) entered forwarding state
[   54.307149] rtl83xx-switch switch@1b000000 lan8: configuring for phy/qsgmii link mode
[   54.410669] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 7 and sds 1 not yet fully implemented
[   54.553766] 8021q: adding VLAN 0 to HW filter on device lan8
[   54.637040] switch: port 8(lan8) entered blocking state
[   54.706019] switch: port 8(lan8) entered disabled state
[   54.775008] rtl83xx-switch switch@1b000000 lan8: entered allmulticast mode
[   54.866668] rtl83xx-switch switch@1b000000 lan8: entered promiscuous mode
[   54.977892] rtl83xx-switch switch@1b000000 lan9: configuring for phy/qsgmii link mode
[   55.081335] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 8 and sds 2 not yet fully implemented
[   55.224546] 8021q: adding VLAN 0 to HW filter on device lan9
[   55.305386] switch: port 9(lan9) entered blocking state
[   55.374344] switch: port 9(lan9) entered disabled state
[   55.443487] rtl83xx-switch switch@1b000000 lan9: entered allmulticast mode
[   55.535254] rtl83xx-switch switch@1b000000 lan9: entered promiscuous mode
[   55.645780] rtl83xx-switch switch@1b000000 lan10: configuring for phy/qsgmii link mode
[   55.750394] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 9 and sds 2 not yet fully implemented
[   55.893918] 8021q: adding VLAN 0 to HW filter on device lan10
[   55.983278] switch: port 10(lan10) entered blocking state
[   56.054576] switch: port 10(lan10) entered disabled state
[   56.125845] rtl83xx-switch switch@1b000000 lan10: entered allmulticast mode
[   56.218480] rtl83xx-switch switch@1b000000 lan10: entered promiscuous mode
[   56.330994] rtl83xx-switch switch@1b000000 lan11: configuring for phy/qsgmii link mode
[   56.435568] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 10 and sds 2 not yet fully implemented
[   56.580079] 8021q: adding VLAN 0 to HW filter on device lan11
[   56.668898] switch: port 11(lan11) entered blocking state
[   56.740174] switch: port 11(lan11) entered disabled state
[   56.811577] rtl83xx-switch switch@1b000000 lan11: entered allmulticast mode
[   56.904325] rtl83xx-switch switch@1b000000 lan11: entered promiscuous mode
[   57.014840] rtl83xx-switch switch@1b000000 lan12: configuring for phy/qsgmii link mode
[   57.119456] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 11 and sds 2 not yet fully implemented
[   57.263772] 8021q: adding VLAN 0 to HW filter on device lan12
[   57.345694] switch: port 12(lan12) entered blocking state
[   57.417025] switch: port 12(lan12) entered disabled state
[   57.488561] rtl83xx-switch switch@1b000000 lan12: entered allmulticast mode
[   57.581737] rtl83xx-switch switch@1b000000 lan12: entered promiscuous mode
[   57.696833] rtl83xx-switch switch@1b000000 lan13: configuring for phy/qsgmii link mode
[   57.801363] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 12 and sds 3 not yet fully implemented
[   57.945768] 8021q: adding VLAN 0 to HW filter on device lan13
[   58.032413] switch: port 13(lan13) entered blocking state
[   58.103597] switch: port 13(lan13) entered disabled state
[   58.174918] rtl83xx-switch switch@1b000000 lan13: entered allmulticast mode
[   58.267642] rtl83xx-switch switch@1b000000 lan13: entered promiscuous mode
[   58.383839] rtl83xx-switch switch@1b000000 lan14: configuring for phy/qsgmii link mode
[   58.488444] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 13 and sds 3 not yet fully implemented
[   58.633598] 8021q: adding VLAN 0 to HW filter on device lan14
[   58.717832] switch: port 14(lan14) entered blocking state
[   58.789115] switch: port 14(lan14) entered disabled state
[   58.860503] rtl83xx-switch switch@1b000000 lan14: entered allmulticast mode
[   58.953343] rtl83xx-switch switch@1b000000 lan14: entered promiscuous mode
[   59.063412] rtl83xx-switch switch@1b000000 lan15: configuring for phy/qsgmii link mode
[   59.167990] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 14 and sds 3 not yet fully implemented
[   59.312406] 8021q: adding VLAN 0 to HW filter on device lan15
[   59.393715] switch: port 15(lan15) entered blocking state
[   59.465008] switch: port 15(lan15) entered disabled state
[   59.536636] rtl83xx-switch switch@1b000000 lan15: entered allmulticast mode
[   59.629334] rtl83xx-switch switch@1b000000 lan15: entered promiscuous mode
[   59.741749] rtl83xx-switch switch@1b000000 lan16: configuring for phy/qsgmii link mode
[   59.846356] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 15 and sds 3 not yet fully implemented
[   59.990817] 8021q: adding VLAN 0 to HW filter on device lan16
[   60.077416] switch: port 16(lan16) entered blocking state
[   60.148678] switch: port 16(lan16) entered disabled state
[   60.220081] rtl83xx-switch switch@1b000000 lan16: entered allmulticast mode
[   60.313127] rtl83xx-switch switch@1b000000 lan16: entered promiscuous mode
[   60.422968] rtl83xx-switch switch@1b000000 lan17: configuring for phy/qsgmii link mode
[   60.527608] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 16 and sds 4 not yet fully implemented
[   60.672696] 8021q: adding VLAN 0 to HW filter on device lan17
[   60.767708] switch: port 17(lan17) entered blocking state
[   60.838988] switch: port 17(lan17) entered disabled state
[   60.910347] rtl83xx-switch switch@1b000000 lan17: entered allmulticast mode
[   61.003195] rtl83xx-switch switch@1b000000 lan17: entered promiscuous mode
[   61.125729] rtl83xx-switch switch@1b000000 lan18: configuring for phy/qsgmii link mode
[   61.230352] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 17 and sds 4 not yet fully implemented
[   61.374700] 8021q: adding VLAN 0 to HW filter on device lan18
[   61.456445] switch: port 18(lan18) entered blocking state
[   61.527804] switch: port 18(lan18) entered disabled state
[   61.599273] rtl83xx-switch switch@1b000000 lan18: entered allmulticast mode
[   61.692564] rtl83xx-switch switch@1b000000 lan18: entered promiscuous mode
[   61.802395] rtl83xx-switch switch@1b000000 lan19: configuring for phy/qsgmii link mode
[   61.907103] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 18 and sds 4 not yet fully implemented
[   62.052212] 8021q: adding VLAN 0 to HW filter on device lan19
[   62.156188] switch: port 19(lan19) entered blocking state
[   62.227420] switch: port 19(lan19) entered disabled state
[   62.298715] rtl83xx-switch switch@1b000000 lan19: entered allmulticast mode
[   62.391792] rtl83xx-switch switch@1b000000 lan19: entered promiscuous mode
[   62.510493] rtl83xx-switch switch@1b000000 lan20: configuring for phy/qsgmii link mode
[   62.615067] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 19 and sds 4 not yet fully implemented
[   62.759662] 8021q: adding VLAN 0 to HW filter on device lan20
[   62.841506] switch: port 20(lan20) entered blocking state
[   62.912756] switch: port 20(lan20) entered disabled state
[   62.984215] rtl83xx-switch switch@1b000000 lan20: entered allmulticast mode
[   63.077668] rtl83xx-switch switch@1b000000 lan20: entered promiscuous mode
[   63.190584] rtl83xx-switch switch@1b000000 lan21: configuring for phy/qsgmii link mode
[   63.295161] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 20 and sds 5 not yet fully implemented
[   63.450092] 8021q: adding VLAN 0 to HW filter on device lan21
[   63.562067] switch: port 21(lan21) entered blocking state
[   63.633302] switch: port 21(lan21) entered disabled state
[   63.704606] rtl83xx-switch switch@1b000000 lan21: entered allmulticast mode
[   63.798077] rtl83xx-switch switch@1b000000 lan21: entered promiscuous mode
[   63.909445] rtl83xx-switch switch@1b000000 lan22: configuring for phy/qsgmii link mode
[   64.013966] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 21 and sds 5 not yet fully implemented
[   64.158580] 8021q: adding VLAN 0 to HW filter on device lan22
[   64.241722] switch: port 22(lan22) entered blocking state
[   64.312983] switch: port 22(lan22) entered disabled state
[   64.384264] rtl83xx-switch switch@1b000000 lan22: entered allmulticast mode
[   64.477825] rtl83xx-switch switch@1b000000 lan22: entered promiscuous mode
[   64.589700] rtl83xx-switch switch@1b000000 lan23: configuring for phy/qsgmii link mode
[   64.694270] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 22 and sds 5 not yet fully implemented
[   64.838996] 8021q: adding VLAN 0 to HW filter on device lan23
[   64.935443] switch: port 23(lan23) entered blocking state
[   65.006695] switch: port 23(lan23) entered disabled state
[   65.077984] rtl83xx-switch switch@1b000000 lan23: entered allmulticast mode
[   65.171120] rtl83xx-switch switch@1b000000 lan23: entered promiscuous mode
[   65.286988] rtl83xx-switch switch@1b000000 lan24: configuring for phy/qsgmii link mode
[   65.391497] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 23 and sds 5 not yet fully implemented
[   65.535953] 8021q: adding VLAN 0 to HW filter on device lan24
[   65.638673] switch: port 24(lan24) entered blocking state
[   65.709866] switch: port 24(lan24) entered disabled state
[   65.781172] rtl83xx-switch switch@1b000000 lan24: entered allmulticast mode
[   65.874407] rtl83xx-switch switch@1b000000 lan24: entered promiscuous mode
[   65.985841] rtl83xx-switch switch@1b000000 lan25: configuring for phy/qsgmii link mode
[   66.090439] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 48 and sds 12 not yet fully implemented
[   66.237232] Realtek RTL8214FC 1b000000.switchcore:mdio-controller-mii:30: power off copper
[   66.349948] Realtek RTL8214FC 1b000000.switchcore:mdio-controller-mii:30: power on fibre
[   66.460591] 8021q: adding VLAN 0 to HW filter on device lan25
[   66.544635] switch: port 25(lan25) entered blocking state
[   66.616011] switch: port 25(lan25) entered disabled state
[   66.687443] rtl83xx-switch switch@1b000000 lan25: entered allmulticast mode
[   66.780948] rtl83xx-switch switch@1b000000 lan25: entered promiscuous mode
[   66.915505] rtl83xx-switch switch@1b000000 lan26: configuring for phy/qsgmii link mode
[   67.020042] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 49 and sds 12 not yet fully implemented
[   67.166483] Realtek RTL8214FC 1b000000.switchcore:mdio-controller-mii:31: power off copper
[   67.283969] Realtek RTL8214FC 1b000000.switchcore:mdio-controller-mii:31: power on fibre
[   67.399158] 8021q: adding VLAN 0 to HW filter on device lan26
[   67.485524] switch: port 26(lan26) entered blocking state
[   67.556835] switch: port 26(lan26) entered disabled state
[   67.628352] rtl83xx-switch switch@1b000000 lan26: entered allmulticast mode
[   67.722031] rtl83xx-switch switch@1b000000 lan26: entered promiscuous mode
[   67.837374] rtl83xx-switch switch@1b000000 lan27: configuring for phy/qsgmii link mode
[   67.941970] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 50 and sds 12 not yet fully implemented
[   68.088167] Realtek RTL8214FC 1b000000.switchcore:mdio-controller-mii:32: power off copper
[   68.198642] Realtek RTL8214FC 1b000000.switchcore:mdio-controller-mii:32: power on fibre
[   68.307283] 8021q: adding VLAN 0 to HW filter on device lan27
[   68.389258] switch: port 27(lan27) entered blocking state
[   68.460484] switch: port 27(lan27) entered disabled state
[   68.531774] rtl83xx-switch switch@1b000000 lan27: entered allmulticast mode
[   68.625213] rtl83xx-switch switch@1b000000 lan27: entered promiscuous mode
[   68.736893] rtl83xx-switch switch@1b000000 lan28: configuring for phy/qsgmii link mode
[   68.841479] realtek-otto-pcs 1b000000.switchcore:pcs: pcs_config(qsgmii) for port 51 and sds 12 not yet fully implemented
[   68.988777] Realtek RTL8214FC 1b000000.switchcore:mdio-controller-mii:33: power off copper
[   69.099269] Realtek RTL8214FC 1b000000.switchcore:mdio-controller-mii:33: power on fibre
[   69.208345] 8021q: adding VLAN 0 to HW filter on device lan28
[   69.290695] switch: port 28(lan28) entered blocking state
[   69.361929] switch: port 28(lan28) entered disabled state
[   69.433398] rtl83xx-switch switch@1b000000 lan28: entered allmulticast mode
[   69.527371] rtl83xx-switch switch@1b000000 lan28: entered promiscuous mode
[   69.632875] rtl83xx_fib_event: FIB_RULE ADD/DEL for IPv6 not supported
[   69.719360] rtl83xx-switch switch@1b000000: add IPv4 route 127.0.0.1/32 (VLAN 0, MAC 00:00:00:00:00:00)
[   69.843509] rtl83xx-switch switch@1b000000: skip loopback/broadcast addresses and default routes
[   69.961530] rtl83xx-switch switch@1b000000: add IPv4 route 127.0.0.0/8 (VLAN 0, MAC 00:00:00:00:00:00)
[   70.084718] rtl83xx-switch switch@1b000000: skip loopback/broadcast addresses and default routes
[   70.247413] rtl83xx-switch switch@1b000000: add IPv4 route 127.255.255.255/32 (VLAN 0, MAC 00:00:00:00:00:00)
[   70.378777] rtl83xx-switch switch@1b000000: skip loopback/broadcast addresses and default routes
```

</details>

Neustradamus commented:
@andyboeh: It is really strange to see: "[    0.000000] MIPS: machine is Zyxel GS1920-24HP" for your GS2210-24LP :/

Neustradamus commented:
@andyboeh: I have seen ending MAC 00...

jonasjelonek commented:
@robimarko @hauke is this fine from your side? Thanks in advance :)

jonasjelonek commented:
> Maybe someone else can test this?

Tried on my GS1920-24HPv2, and doesn't work. The button is attached to the RTL8231 which doesn't support interrupts. Thus, it fails during boot with:
```
gpio-keys keys: failed to get irq for gpio:-2
```

Please change back to `gpio-keys-polled`.

andyboeh committed:
realtek: add support for ZyXEL GS1920-24HPv1

The GS1920-24HPv1 is a switch with 24 copper ports and 4 combo SFP/copper
ports and PoE on the first 24 ports.

Specifications:
---------------
  * SoC: Realtek RTL8292M
  * Flash: 16 MiB SPI flash
  * RAM: 128 MiB
  * Ethernet: 24x 10/100/1000 Mbps
  * Buttons: 1x "Reset" button
  * UART: 1x serial header, standard DCE pinout (Tx = 2, Rx = 3, Gnd = 5);
          9600 baud, 8n1, +- 5.6V logic levels
  * SFP: 4 combo copper/SFP ports
  * PoE: 24x
  * Fans: ADT7468 fan controller

Works:
------
  - (24) RJ-45 ethernet ports
  - Switch functions
  - Buttons
  - LEDs (partial support, the wrong LEDs light up)
  - Manual fan control

Not yet enabled:
----------------
  - PoE (requires patches to realtek-poe to support i2c)
  - Combo ports (link is up, but no data is transferred)

Fans:
-----
After boot, the fans are running in full speed mode. You can interact
with the fan controller at /sys/class/hwmon/

Installation:
-------------

This device uses ZyNOS instead of Linux, this makes installation a bit
more cumbersome. Serial console is required!

1. Set the switch to boot from the first image. This step is crucial,
   it will fail to boot if this is not set properly.

2. Connect to the switch using serial and interrupt the boot process
   to enter debug/recovery mode.

3. Load the OpenWrt initramfs image via XMODEM. You need to obtain an
   unlock code, based on your MAC address, first. See the excellent write
   up at https://www.ixo.de/info/zyxel_uclinux/ for details. Replace
   unlock_code in the commands below by the code obtained.
   After running ATBA5, the terminal needs to be closed and re-opened
   with 115200 baud. This speeds up the file transfer significantly!
   The file length in bytes need to be given instead of file_length below.
   You also need an XMODEM upload utility like "lrzsz-sx -X" to transfer
   the file. Start the XMODEM upload after running the ATUPxxxx command:

     > ATEN1,unlock_code
     > ATBA5
     > ATUP80100000,file_length
     > ATGO80100000

4. Wait for OpenWrt to boot. Once this is done, transfer the loader binary
   and the sysupgrade image to "/tmp" using scp.

5. Install OpenWrt permanently by running the following two commands on
   the switch (over SSH):

    > mtd write /tmp/loader.bin loader
    > mtd write /tmp/sysupgrade.bin firmware

6. Reboot the switch and enjoy OpenWrt.

NB: You do not need to touch the loader binary unless it's recommended.
    The loader is not part of a regular sysupgrade file and will be left
    untouched. The boot loader only checks if the loader is valid to be
    able to boot.

Recovery/ Return to stock:
--------------------------

Just spam the "u" key during (or "z" for 9600 baud) during memory testing
to trigger a recovery XMODEM upload at 115200 baud. A standard OEM upgrade
image works properly.

Signed-off-by: Andreas Böhler <dev@aboehler.at>
Link: https://github.com/openwrt/openwrt/pull/20439
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

andyboeh commented:
@plappermaul @jonasjelonek 
Thanks for the pings and sorry that it took so long - real life is keeping me very busy. This should be good to go now!

jonasjelonek reviewed (approved):
Thanks for updating, looks good to me.

plappermaul reviewed (approved):
Thank you.
